packages/notifier/test/test-publish-kit.js

```diff
@@ -174,10 +177,10 @@ const verifySubscribeAfter = test.macro(async (t, makePublishKit) => {
   const { publisher, subscriber } = /** @type {makePublishKit} */ (
     makePublishKit
   )();
-  for (const badCount of [1n, 0, '', false, Symbol('symbol'), {}]) {
-    t.throws(
+  for await (const badCount of [1n, 0, '', false, Symbol('symbol'), {}]) {
+    await t.throwsAsync(
       // @ts-ignore deliberate invalid arguments for testing
-      () => subscriber.subscribeAfter(badCount),
+      () => E(subscriber).subscribeAfter(badCount),
```

Comment:
the check is synchronous https://github.com/Agoric/agoric-sdk/blob/6bd032ff2c9cc7955e8da5a88f4da10bb48dcc8e/packages/notifier/src/publish-kit.js#L127-L128

do you mean to only guarantee that check will happen when the promise is settled?

if not, wouldn't it work to leave this loop be synchronous using an `E(subscriber)` presence?

Reply:
Again, thanks for catching this.  I forgot to roll back some changes.  I've now carefully audited the `test/*` changes and made sure they still make sense.